src/connector/src/parser/dynamodb/cdc_json_parser.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use assert_matches::assert_matches;
+use risingwave_common::types::{DataType, DatumCow, ScalarRefImpl, ToDatumRef};
+
+use crate::error::ConnectorResult;
+use crate::only_parse_payload;
+use crate::parser::dynamodb::{build_dynamodb_json_accessor_builder, map_rw_type_to_dynamodb_type};
+use crate::parser::unified::util::apply_row_operation_on_stream_chunk_writer;
+use crate::parser::unified::{ChangeEvent, ChangeEventOperation};
+use crate::parser::{
+    Access, AccessBuilderImpl, AccessError, ByteStreamSourceParser, ParserFormat,
+    SourceStreamChunkRowWriter, SpecificParserConfig,
+};
+use crate::source::{SourceColumnDesc, SourceContext, SourceContextRef};
+
+// See https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_Record.html
+// An example of a DynamoDB change event from Kinesis:
+// {
+//     "awsRegion":"us-east-1",
+//     "eventID":"26b1b865-1ee7-47f9-816e-d51cdeef9dc5",
+//     "eventName":"MODIFY",
+//     "userIdentity":null,
+//     "recordFormat":"application/json",
+//     "tableName":"wkx-test-orders",
+//     "dynamodb":{
+//         "ApproximateCreationDateTime":1720046486486729,
+//         "Keys":{
+//             "customer_name":{
+//                 "S":"Bob Lee"
+//             },
+//             "order_id":{
+//                 "N":"3"
+//             }
+//         },
+//         "NewImage":{
+//             "order_status":{
+//                 "N":"3"
+//             },
+//             "order_date":{
+//                 "N":"1720046486"
+//             },
+//             "order_id":{
+//                 "N":"3"
+//             },
+//             "price":{
+//                 "N":"63.06"
+//             },
+//             "product_id":{
+//                 "N":"2060"
+//             },
+//             "customer_name":{
+//                 "S":"Bob Lee"
+//             }
+//         },
+//         "OldImage":{
+//             "order_status":{
+//                 "N":"3"
+//             },
+//             "order_date":{
+//                 "N":"1720037677"
+//             },
+//             "order_id":{
+//                 "N":"3"
+//             },
+//             "price":{
+//                 "N":"63.06"
+//             },
+//             "product_id":{
+//                 "N":"2060"
+//             },
+//             "customer_name":{
+//                 "S":"Bob Lee"
+//             }
```

Comment:
Is there a chance that they just keep the fields in `Keys` instead of the whole row?

Reply:
They don't seem provide this option.